source/SIL.AppBuilder.Portal.Frontend/src/data/containers/with-current-user/index.tsx

```diff
@@ -0,0 +1,53 @@
+import * as React from 'react';
+
+import { ICurrentUserProps } from './types';
+import { withFetcher, withCurrentUser as hoc } from './hoc';
+
+const CurrentUserContext = React.createContext<ICurrentUserProps>({
+  currentUser: undefined,
+  currentUserProps: {}
+});
+
+export const Provider = withFetcher()(
+  (props: ICurrentUserProps & { children: any }) => {
+    const { currentUser, currentUserProps, children } = props;
+
+    return (
+      <CurrentUserContext.Provider value={{ currentUser, currentUserProps }}>
+        {children}
+      </CurrentUserContext.Provider>
+    );
+});
+
+
+export function withCurrentUserContext(InnerComponent) {
```

Comment:
ahahhhhahahahahahahahahaaaaahhhhhhh.   nice.

![unicorn.gif](https://s3.amazonaws.com/dashcode.snippet.images/unicorn.gif)

Reply:
like, i think i have a way to crack this signal-r thing now and i understand the idea of using a context to do it.

Reply:
yay!